src/devices/sound/mas3507d.cpp

```diff
@@ -226,14 +252,151 @@ void mas3507d_device::i2c_device_got_stop()
 	logerror("MAS I2C: got stop\n");
 }
 
+int gain_to_db(int val) {
```

Comment:
Urgh, that's a direct multiplier.  In other words, db = 20*log10((0x100000 - v)/0x80000)